rfcs/experimental/flow_graph_wait_for_one/README.md

```diff
@@ -0,0 +1,481 @@
+
+Once the element and the metainfo are stored in the buffer, the node will try to push them to the successor. If one of the successors accepts the message and the metainfo,
+both of them are removed from the buffer. Otherwise, the push-pull protocol works and the successor should return for the item once it becomes available by calling
+``try_get(msg, metainfo)`` or ``try_reserve(msg, metainfo)``.
+
+Since placing the buffers before rejecting nodes is not the only use-case, there is a risk of issues relates to buffering. It is described in details in the [separate section](#buffering-the-metainfo).
+
+### Broadcast-push buffering nodes
+
+The issue with broadcast-push ``overwrite_node`` and ``write_once_node`` is these nodes stores the received item and even if this item is accepted by one of the successors,
+it would be broadcasted to others and kept in the buffer.
+
+Since the metainformation is kept in the buffer together with the message itself, even if the computation is completed, the ``try_put_and_wait`` would stuck because of the reference
+held by the buffer.
+
+Even the ``wait_for_all()`` call would be able to finish in this case since it counting only the tasks in progress and ``try_put_and_wait`` would still be blocked.
+
+``try_put_and_wait`` feature for the graph containing these nodes should be used carefully because of this issue:
+
+* The ``overwrite_node`` should be explicitly reset by calling ``node.reset()`` or the element with the stored metainfo should be overwritten with another element.
+* The ``write_once_node`` should be explicitly reset by calling ``node.reset()`` since the item cannot be overwritten.
+
+### ``broadcast_node``
+
+While ``broadcast_node::try_put_task`` is called with the metainfo argument - both item and the associated metainformation would be broadcasted to each successor of the node.
+
+### ``limiter_node``
+
+If the threshold of the node was not reached, both value and the metainformation should be broadcasted to the successors. Otherwise- both should be rejected and buffered by another node. 
+
+Metainformation on the decrement port is ignored since this signal should not be considered part of working on the original message.
+
+### Queueing ``join_node``
+
+Each input port of the join_node should support the queue for both values and the associated metainformations. Once all of the input ports would contain the value, the values
+should be combined into single tuple output and the metainformation objects should be combined into single metainfo using `metainfo1.merge(metainfo2)`, associated with the tuple
+and submitted to successors.
+
+If the item with the metainformation is stored in the internal queue of one of the input ports, but items never received by other ports, the item and the metainformation would be kept in the
+queue and block the corresponding ``try_put_and_wait`` call.
```

Comment:
```suggestion
If an item with the metainformation is stored in the internal queue of one of the input ports, but items are never received by other ports, the item and the metainformation will be kept in the
queue and block the corresponding ``try_put_and_wait`` call.
```

Reply:
We may need to add a `clear` function to `join_node` to allow explicit clearing of the internal queues.